test/System.IdentityModel.Tokens.Jwt.Tests/JwtSecurityTokenHandlerTests.WithContextSwitches.cs

```diff
@@ -0,0 +1,117 @@
+                TokenType = "JWE"
+            };
+
+            JwtSecurityTokenHandler tokenHandler = new JwtSecurityTokenHandler();
+            JwtSecurityToken token = tokenHandler.CreateJwtSecurityToken(tokenDescriptor);
+
+            Assert.NotNull(token);
+            Assert.NotEqual(token.Header.Alg, SecurityAlgorithms.RsaOAEP);
+            Assert.Equal(token.Header.Alg, SecurityAlgorithms.RsaOaepKeyWrap);
+        }
+
+        [Fact]
+        public void JsonWebTokenHandler_CreateToken_AddShortFormMappingForRsaOAEPEnabled()
+        {
+            AppContext.SetSwitch(X509EncryptingCredentials._useShortNameForRsaOaepKey, true);
+            var encryptingCredentials = new X509EncryptingCredentials(Default.Certificate);
+            SecurityTokenDescriptor tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Issuer = Default.Issuer,
+                IssuedAt = DateTime.UtcNow.Subtract(new TimeSpan(1, 0, 0)),
+                Subject = new ClaimsIdentity(Default.PayloadJsonClaims),
+                NotBefore = DateTime.UtcNow.Subtract(new TimeSpan(1, 0, 0)),
+                Expires = DateTime.UtcNow.Subtract(new TimeSpan(0, 10, 0)),
+                SigningCredentials = Default.AsymmetricSigningCredentials,
+                EncryptingCredentials = encryptingCredentials,
+                TokenType = "JWE"
+            };
+
+            JsonWebTokenHandler tokenHandler = new JsonWebTokenHandler();
+            var token = tokenHandler.CreateToken(tokenDescriptor);
+            JsonWebToken jsonToken = tokenHandler.ReadToken(token) as JsonWebToken;
+
+            Assert.NotNull(jsonToken);
+            Assert.NotEqual(jsonToken._alg, SecurityAlgorithms.RsaOaepKeyWrap);
+            Assert.Equal(jsonToken._alg, SecurityAlgorithms.RsaOAEP);
+        }
+
+        [Fact]
+        public void JsonWebTokenHandler_CreateToken_AddShortFormMappingForRsaOAEPDisabled()
+        {
+            AppContext.SetSwitch(X509EncryptingCredentials._useShortNameForRsaOaepKey, false);
+            var encryptingCredentials = new X509EncryptingCredentials(Default.Certificate);
+            SecurityTokenDescriptor tokenDescriptor = new SecurityTokenDescriptor
+            {
+                Issuer = Default.Issuer,
+                IssuedAt = DateTime.UtcNow.Subtract(new TimeSpan(1, 0, 0)),
+                Subject = new ClaimsIdentity(Default.PayloadJsonClaims),
+                NotBefore = DateTime.UtcNow.Subtract(new TimeSpan(1, 0, 0)),
+                Expires = DateTime.UtcNow.Subtract(new TimeSpan(0, 10, 0)),
+                SigningCredentials = Default.AsymmetricSigningCredentials,
+                EncryptingCredentials = encryptingCredentials,
+                TokenType = "JWE"
+            };
+
+            JsonWebTokenHandler tokenHandler = new JsonWebTokenHandler();
+            var token = tokenHandler.CreateToken(tokenDescriptor);
+            JsonWebToken jsonToken = tokenHandler.ReadToken(token) as JsonWebToken;
```

Comment:
You could write: JsonWebToken jsonToken = new JsonWebToken(token);

Reply:
Done. Thanks Brent!